modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -154,14 +171,19 @@ class GmosControllerEpics[T<:GmosController.SiteDependentTypes](encoders: GmosCo
 
   private val PixelsToMicrons = 15.0
 
-  def setCCConfig(cc: GmosController.Config[T]#CCConfig): SeqAction[Unit] = for {
-    _ <- setFilters(cc.filter)
-    _ <- setDisperser(cc.disperser)
-    _ <- setFPU(cc.fpu)
-    _ <- CC.setStageMode(encoders.stageMode.encode(cc.stage))
-    _ <- CC.setDtaXOffset(cc.dtaX.intValue.toDouble*PixelsToMicrons)
-    _ <- cc.useElectronicOffset.fold(CC.setElectronicOffsetting(0))(e => CC.setElectronicOffsetting(encode(e)))
-  } yield ()
+  def setCCConfig(cc: GmosController.Config[T]#CCConfig): SeqAction[Unit] = {
+    val stage = encoders.stageMode.encode(cc.stage)
+
+    for {
+      _ <- setFilters(cc.filter)
+      _ <- setDisperser(cc.disperser)
+      _ <- setFPU(cc.fpu)
+      _ <- smartSetParam(stage, GmosEpics.instance.stageMode, CC.setStageMode(stage))
+      _ <- CC.setDtaXOffset(cc.dtaX.intValue.toDouble*PixelsToMicrons)
+      _ <- cc.useElectronicOffset.fold(CC.setElectronicOffsetting(0))(e =>
```

Comment:
Perhaps a clearer constant would be better than the `0` in `setElectronicOffseting`